### DISCUSSION TITLE
Question about reverse mapping from `key` to `slice`.

### DISCUSSION THREAD

**Discussion by JoOkuma** [zarr-developers/zarr-python#987] [Q&A]
Hi, I'm not sure what's the best place to ask `zarr` questions so I'm posting this here.

I'm trying to iterate as fast as possible over a `zarr.Array` with very few chunks initialized.
I couldn't find a method for this so I'm implementing my own function that will return a sequence of chunks and their respective slices.

I started with this function to get the chunks' key:

```python3
def get_initialized_keys(array: Array):
    # key pattern for chunk keys
    prog = re.compile(r'\.'.join([r'\d+'] * min(1, array.ndim)))

    # yield chunk keys
    for k in listdir(array.chunk_store, array._path):
        if prog.match(k):
            yield k
```

Is there any API to get the reverse `BasicIndexer` behavior, so for a given `key` (`chunk_selection`) I would get its slice (`out_selection`)?

Thanks

**Comment by sanketverma1704** [comment]
Hi @JoOkuma.

Thanks for posting your question.

> I'm not sure what's the best place to ask `zarr` questions

As it happens we have just opened [discussions](https://github.com/zarr-developers/zarr-python/discussions) section in zarr-python and it'd be great if you could post your question there!

Thanks!

**Reply by joshmoore** [reply]
Migrated.

**Comment by jakirkham** [comment]
Maybe worth reading over issue ( https://github.com/zarr-developers/zarr-python/issues/518 )
